bindings/python/ethash/__init__.py

```diff
@@ -1,21 +1,39 @@
 # ethash: C/C++ implementation of Ethash, the Ethereum Proof of Work algorithm.
 # Copyright 2019 Pawel Bylica.
+# Copyright 2023 Sam Wilson.
 # Licensed under the Apache License, Version 2.0.
 
-from _ethash import ffi, lib
+from _ethash import ffi, lib  # type: ignore
 
+from typing import TYPE_CHECKING, Tuple
 
-def keccak_256(data):
+from collections.abc import Sized
+
+if TYPE_CHECKING:
+    from typing import Protocol
+
+    from _typeshed import WriteableBuffer, ReadableBuffer
+
+    class SizedWriteableBuffer(Protocol, WriteableBuffer, Sized):
+        pass
+
+    class SizedReadableBuffer(Protocol, ReadableBuffer, Sized):
+        pass
+
+
+def keccak_256(data: "SizedWriteableBuffer") -> bytes:
     hash = lib.ethash_keccak256(ffi.from_buffer(data), len(data))
```

Comment:
_unrelated nit:_ `hash` is a reserved Python word. This could be mitigated internally in a function by calling it `_hash` or `keccak256_hash`. If unbothered by it, feel free to ignore.

Reply:
Let's change it to `h`.